### DISCUSSION TITLE
Dev mode 'blank screen', any work around?

### DISCUSSION THREAD

**Discussion by OJFord** [tauri-apps/tauri#1233] [Q&A] [answered]
Hi, I'm not opening an issue because I get the impression (#1212 #1092 #808 #448 #1140) this is well-known, 'fixed in the next release', and probably caused by a number of underlying issues.

What's not clear to me though is if there's any work-around in the mean time? (Other than a full debug build I mean, which works fine.)

I did have it working, and I think it broke when a dependency updated (in which Tauri 0.11.0 -> 0.11.1, but reverting that didn't fix it), now it only works in `tauri build [--debug]`, not `tauri dev`, which just gives me a 'white screen', no HTML, and (possibly unrelatedly) opening the dev tools warns:

> NSWindow warning: adding an unknown subview: <WKInspectorWKWebView [...]

My Tauri conf hasn't changed since it was working, and is just what I got from running init with v0.11.0 (incl. embedded-server enabled, as is a suggested fix in at least one of the issues linked above).

<details>
<summary>tauri info</summary>

```
[tauri]: running info

Operating System - Darwin(20.2.0) - darwin/x64

Node.js environment
  Node.js - 15.8.0
  tauri.js - 0.14.1

Rust environment
  rustc - 1.50.0
  cargo - 1.50.0
  tauri-bundler - 0.9.4

Global packages
  NPM - 7.5.0
  yarn - 1.22.10

App directory structure
/.git
/dist
/node_modules
/src-tauri

App
  tauri.rs - 0.11.1
  mode - embedded-server
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: http: https: 'unsafe-eval' 'unsafe-inline'
  distDir - ../dist
  devPath - http://localhost:4000
✨  Done in 3.43s.
```

</details>

Many thanks!

**Comment by nothingismagick** [comment, marked as answer]
Hmm. There are soooo many things that this could be, and I wonder if it isn't about your source dist. Can you confirm that a very basic example also fails similarly on your setup?

**Reply by OJFord** [reply]
Hm, I've tracked it down to a seemingly innocuous change of adding  `--quiet` flag to `sirv` which I'm using serve `dist` on `yarn dev`.
The really curious thing is that I can add it back as long it's after `dist`. That is, with `sirv --dev --port=4000 --quiet dist` nothing is served (tested in browser); with `sirv --dev --port=4000 dist --quiet` it works fine!

(Found it by removing it to see if there was some error/warning it was hiding, but instead it fixed it...)

Sorry for wasting your time.

**Reply by nothingismagick** [reply]
interesting! Thanks for the follow up. Yeah, its definitely one of the things that we struggle with - because of our agnostic approach, any mistake anywhere (even upstream or unintentional) can make it seem like Tauri itself is broken... :(

**Reply by OJFord** [reply]
Well I'm really liking it so far 🙂

I wonder though if there's a reason that the `tauri dev` can't use the embedded server too? It'd remove some scope for errors like mine, remove differences where various different tools (sirv, python -m http.server, whatever else) are used for the same thing, and reduce differences between 'dev' and 'prod' (or at least 'dev' and a debug build).

**Reply by nklayman** [reply]
You can pass a folder to `build.devPath` in `tauri.conf.json` and it will load the data from there. However, this isn't recommended because instead of HMR, it will have to rebuild your app every time you change it. Most people want to use their own webpack-dev-server or similar, so we don't include our own.

**Reply by OJFord** [reply]
Oh nice ok, thanks! (For now at least) my build is actually quick enough that that might be fine. Cheers.

**Reply by nklayman** [reply]
Unfortunately, there is a regression in the next version of Tauri that causes builds (even for just a hello world app) to take nearly 10 seconds. Because of this, I'd suggest still using another file server and pointing Tauri at that.
